tests/unit/test_system.py

```diff
@@ -156,3 +164,41 @@ def test_unattended_upgrades(self):
                'upgrades.')
         issues = list(IssuesStore().load().values())[0]
         self.assertEqual([issue['desc'] for issue in issues], [msg])
+
+    @mock.patch('hotsos.core.plugins.kernel.KernelBase')
+    @mock.patch('hotsos.core.host_helpers.CLIHelper')
+    @mock.patch('hotsos.core.ycheck.engine.YDefsLoader._is_def',
+                new=utils.is_def_filter('amd_iommu_pt.yaml'))
+    def test_amd_iommu_pt_fail(self, mock_cli, mock_kernel):
+        mock_cli.return_value = mock.MagicMock()
+        mock_cli.return_value.lscpu.return_value = \
+            FAKE_AMD_LSCPU.splitlines(keepends=True)
+
+        mock_kernel.return_value = mock.MagicMock()
+        mock_kernel.return_value.boot_parameters = \
+            ['intel_iommu=on']
+
+        YScenarioChecker()()
+        msg = ('This host is using an AMD AMD EPYC 7502 32-Core Processor cpu '
+               'but is not using iommu passthrough mode (e.g. set iommu=pt in '
+               'boot parameters) which is recommended in order to get the '
+               'best performance e.g. for networking.')
+        issues = list(IssuesStore().load().values())[0]
+        self.assertEqual([issue['desc'] for issue in issues], [msg])
+
+    @mock.patch('hotsos.core.plugins.kernel.KernelBase')
+    @mock.patch('hotsos.core.host_helpers.CLIHelper')
+    @mock.patch('hotsos.core.ycheck.engine.YDefsLoader._is_def',
+                new=utils.is_def_filter('amd_iommu_pt.yaml'))
+    def test_amd_iommu_pt_pass(self, mock_cli, mock_kernel):
+        mock_cli.return_value = mock.MagicMock()
+        mock_cli.return_value.lscpu.return_value = \
+            FAKE_AMD_LSCPU.splitlines(keepends=True)
+
+        mock_kernel.return_value = mock.MagicMock()
+        mock_kernel.return_value.boot_parameters = \
+            ['intel_iommu=on', 'iommu=pt']
```

Comment:
does this mean both intel_iommu and iommu must be set to on and pt?
I am only sure for AMD EPYC series CPU, iommu must be set to pt, but not sure about intel_iommu

Reply:
hi @gerald-yang-tw no only iommu=pt is needed:

  iommu_not_pt:
    property:
      path: hotsos.core.plugins.kernel.KernelBase.boot_parameters
      ops: [[contains, 'iommu=pt'], [not_]]

I just put both to test for false positives.

Reply:
Got it, thanks Ed